### PR TITLE
Map system state API to login filter

### LINKED ISSUE #5893: Adapt CLI to use keycloak openid authentication
We need to change this code https://github.com/eclipse/che/blob/master/dockerfiles/lib/src/api/wsmaster/auth/auth-data.ts#L129 to use keycloak openid authentication.

1. I think we need a new client in che realm. Let's say 'curl' with "Bearer only" authentification. 
2. Then, change the way how cli obtain token to something like this. ```curl --data "grant_type=password&client_id=che-public&username=admin&password=123456" http://192.168.65.2:5050/auth/realms/che/protocol/openid-connect/token```

See also http://blog.keycloak.org/2015/10/getting-started-with-keycloak-securing.html

### PR DESCRIPTION
**Pull request by garagatyi**
### What does this PR do?
It is needed since it is secured by permissions system which requires
an authenticated user.
Also, use base API endpoint for checking Che in OpenShift deploy script since system state API which was used before is secured now.

### What issues does this PR fix or reference?
Fixes #5893

### CODE REVIEW

**Review by skabashnyuk — APPROVED** [review]

File: dockerfiles/init/modules/openshift/files/scripts/wait_until_che_is_available.sh [hunk 1]
```diff
@@ -44,7 +44,7 @@ elif [ ${SECONDS} -lt ${end} ]; then
   exit 1
 fi
 
-che_http_status=$(curl -s -o /dev/null -I -w "%{http_code}" "${CHE_API_ENDPOINT}/system/state")
+che_http_status=$(curl -s -o /dev/null -I -w "%{http_code}" "${CHE_API_ENDPOINT}/")
```
Comment (benoitf) [inline dockerfiles/init/modules/openshift/files/scripts/wait_until_che_is_available.sh:47]: @garagatyi it's strange. It means that you return `OK` before system is really available.

Comment (benoitf) [reply dockerfiles/init/modules/openshift/files/scripts/wait_until_che_is_available.sh:47]: Also OPENSHIFT_TOKEN is a token to access secure endpoint
https://github.com/garagatyi/che/blob/02252426f8e5ef5bb687ca9592573470e9ea866e/dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh#L182-L188

Comment (garagatyi) [reply dockerfiles/init/modules/openshift/files/scripts/wait_until_che_is_available.sh:47]: I think it is OpenShift token. Does it work in Che?

Comment (benoitf) [reply dockerfiles/init/modules/openshift/files/scripts/wait_until_che_is_available.sh:47]: AFAIK we're in the che for openshift script there

Comment (garagatyi) [reply dockerfiles/init/modules/openshift/files/scripts/wait_until_che_is_available.sh:47]: I think that checking /api is pretty the same check, so it's ok to secure this one

Comment (benoitf) [reply dockerfiles/init/modules/openshift/files/scripts/wait_until_che_is_available.sh:47]: it's not. If someone has stopped the system you would think system is ready while it is not.

Comment (garagatyi) [reply dockerfiles/init/modules/openshift/files/scripts/wait_until_che_is_available.sh:47]: CLI just checks whether the response is 200 or not. System state JSON data are not checked.  It is equal to the check that API responds.

Comment (benoitf) [reply dockerfiles/init/modules/openshift/files/scripts/wait_until_che_is_available.sh:47]: so you've found a bug in this script ;-)

Comment (garagatyi) [reply dockerfiles/init/modules/openshift/files/scripts/wait_until_che_is_available.sh:47]: Well as far as I understand system state API returns 403 in che-multiuser branch now. Permissions are checked but the user is anonymous, so it should not work in any case.

**Review by l0rd — APPROVED** [review]

**Comment by garagatyi** [comment]
@benoitf Do you think something specific should be changed in this PR?

**Comment by benoitf** [comment]
@garagatyi I think but l0rd is owner of the script so it was up to him

**Review by riuvshin — APPROVED** [review]

**Pull request merged**